test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts

```diff
@@ -530,4 +530,179 @@ contract('DCAStrategiesPermissionsHandler', () => {
       chainId: BigNumber;
     };
   });
+
+  describe('permissionPermit', () => {
+    const TOKEN_ID = 1;
+    const OPERATOR = wallet.generateRandomAddress();
+    let owner: Wallet, stranger: Wallet;
+
+    given(async () => {
+      owner = await wallet.generateRandom();
+      stranger = await wallet.generateRandom();
+      await DCAStrategiesPermissionsHandlerMock.mint(owner.address, []);
+    });
+
+    function permitTest({ when: title, permissions }: { when: string; permissions: Permission[] }) {
+      when(title, () => {
+        let tx: TransactionResponse;
+
+        given(async () => {
+          tx = await signAndPermit({ signer: owner, permissions: [{ operator: OPERATOR, permissions }] });
+        });
+
+        then('operator gains permissions', async () => {
+          for (const permission of permissions) {
+            expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OPERATOR, permission)).to.be.true;
+          }
+        });
+
+        then('nonces is increased', async () => {
+          expect(await DCAStrategiesPermissionsHandlerMock.nonces(owner.address)).to.be.equal(1);
+        });
+
+        then('event is emitted', async () => {
+          const id = await readArgFromEventOrFail(tx, 'Modified', 'tokenId');
+          const emittedPermissions: any = await readArgFromEventOrFail(tx, 'Modified', 'permissions');
+          expect(id).to.equal(TOKEN_ID);
+          expect(emittedPermissions.length).to.equal(1);
+          expect(emittedPermissions[0].operator).to.equal(OPERATOR);
+          expect(emittedPermissions[0].permissions).to.eql(permissions);
+        });
+      });
+    }
+
+    permitTest({
+      when: `setting only one permission`,
+      permissions: [Permission.INCREASE],
+    });
+
+    permitTest({
+      when: `setting two permissions`,
+      permissions: [Permission.REDUCE, Permission.TERMINATE],
+    });
+
+    permitTest({
+      when: `setting three permissions`,
```

Comment:
Can you add a test that sets four permissions? Just to cover all cases hehe

Reply:
Done sir

Reply:
Merge away, the PR is already approved 😄